NoClassDefFoundError: io/micronaut/runtime/server/EmbeddedServerInstance when using gRPC in Micronaut 2.4.0-SNAPSHOT

An application with gRPC server using Micronaut 2.4.0-SNAPSHOT doesn't start because there is a missing class `io.micronaut.runtime.server.EmbeddedServerInstance`.

In 2.3.x the class `EmbeddedServerInstance` is in `micronaut-runtime` module in the package `io.micronaut.runtime.server`. 
In 2.4.x the class still is in `micronaut-runtime` module but in `io.micronaut.discovery` package.

P.S: This also happens with the Consul test application with a similar error message (class missing `io.micronaut.runtime.server.EmbeddedServerInstance`)


### Steps to Reproduce

- `git clone https://github.com/micronaut-graal-tests/micronaut-grpc-graal`
- `cd micronaut-grpc-graal`
- `git checkout 2.4.x`
- `./gradlew server:run`

### Expected Behaviour

The application should start

### Actual Behaviour

The application doesn't start:

```
 __  __ _                                  _   
|  \/  (_) ___ _ __ ___  _ __   __ _ _   _| |_ 
| |\/| | |/ __| '__/ _ \| '_ \ / _` | | | | __|
| |  | | | (__| | | (_) | | | | (_| | |_| | |_ 
|_|  |_|_|\___|_|  \___/|_| |_|\__,_|\__,_|\__|
  Micronaut (v2.4.0-SNAPSHOT)

13:30:08.638 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: io/micronaut/runtime/server/EmbeddedServerInstance
java.lang.NoClassDefFoundError: io/micronaut/runtime/server/EmbeddedServerInstance
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        at io.micronaut.grpc.server.GrpcEmbeddedServer.lambda$start$1(GrpcEmbeddedServer.java:174)
        at java.util.Optional.ifPresent(Optional.java:159)
        at io.micronaut.grpc.server.GrpcEmbeddedServer.start(GrpcEmbeddedServer.java:162)
        at io.micronaut.grpc.server.GrpcEmbeddedServer.start(GrpcEmbeddedServer.java:60)
        at io.micronaut.runtime.Micronaut.lambda$start$2(Micronaut.java:77)
        at java.util.Optional.ifPresent(Optional.java:159)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
        at example.micronaut.Application.main(Application.java:8)
Caused by: java.lang.ClassNotFoundException: io.micronaut.runtime.server.EmbeddedServerInstance
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        ... 22 common frames omitted
```

### Environment Information

- **Operating System**: Linux Mint 20.1
- **Micronaut Version:** 2.4.0-SNAPSHOT
- **JDK Version:** 8

--- a/runtime/src/main/java/io/micronaut/runtime/server/EmbeddedServerInstance.java
+++ b/runtime/src/main/java/io/micronaut/runtime/server/EmbeddedServerInstance.java
@@ -1,0 +1,24 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.runtime.server;
+
+/**
+ * @deprecated Use {@link io.micronaut.discovery.EmbeddedServerInstance} instead.
+ * @author graemerocher
+ */
+@Deprecated
+public interface EmbeddedServerInstance extends io.micronaut.discovery.EmbeddedServerInstance {
+}

